src/sentry/tasks/reports.py

```diff
@@ -299,15 +339,140 @@ def prepare_project_usage_summary((start, stop), project):
     )
 
 
-def prepare_project_report(interval, project):
-    return (
-        prepare_project_series(interval, project),
-        prepare_project_aggregates(interval, project),
-        prepare_project_issue_summaries(interval, project),
-        prepare_project_release_list(interval, project),
-        prepare_project_usage_summary(interval, project),
+def get_calendar_range((_, stop_time), months):
+    assert (
+        stop_time.hour,
+        stop_time.minute,
+        stop_time.second,
+        stop_time.microsecond,
+        stop_time.tzinfo,
+    ) == (0, 0, 0, 0, pytz.utc)
+
+    last_day = stop_time - timedelta(days=1)
+
+    stop_month_index = month_to_index(
+        last_day.year,
+        last_day.month,
+    )
+
+    start_month_index = stop_month_index - months + 1
+    return start_month_index, stop_month_index
+
+
+def get_calendar_query_range(interval, months):
+    start_month_index, _ = get_calendar_range(interval, months)
+
+    start_time = datetime(
+        day=1,
+        tzinfo=pytz.utc,
+        *index_to_month(start_month_index)
     )
 
+    return start_time, interval[1]
+
+
+def clean_calendar_data(project, series, start, stop, rollup, timestamp=None):
+    earliest = tsdb.get_earliest_timestamp(rollup, timestamp=timestamp)
+
+    def remove_invalid_values(item):
+        timestamp, value = item
+        if timestamp < earliest:
+            value = None
+        elif to_datetime(timestamp) < project.date_added:
+            value = None
+        return (timestamp, value)
+
+    return map(
+        remove_invalid_values,
+        clean_series(
+            start,
+            stop,
+            rollup,
+            series,
+        ),
+    )
+
+
+def prepare_project_calendar_series(interval, project):
+    start, stop = get_calendar_query_range(interval, 3)
+
+    rollup = 60 * 60 * 24
+    series = tsdb.get_range(
+        tsdb.models.project,
+        [project.id],
+        start,
+        stop,
+        rollup=rollup,
+    )[project.id]
+
+    return clean_calendar_data(
+        project,
+        series,
+        start,
+        stop,
+        rollup,
+    )
+
+
+def build(name, fields):
```

Comment:
This is a good idea (and what I had previously intended to do, just never did) with terrible naming and a debatably dodgy implementation.